lib/features/map_view/widgets/bottom_scroll_sheet/data_list.dart

```diff
@@ -16,19 +17,21 @@ class DataSliverList<T extends GoogleNavigable> extends ConsumerWidget {
   Widget build(BuildContext context, WidgetRef ref) {
     final itemsState = ref.watch(context.mapDataController<T>());
     return switch (itemsState) {
-      AsyncLoading() => const DataListLoading(),
+      // misz-masz here fixed flickering
+      // we probably should consider similar refactoring to all switches, but won't be visible in our other use-cases
+      AsyncValue(:final IList<T?> value) =>
+        _DataSliverList<T>(value.whereNonNull.toIList()),
       AsyncError(:final error) =>
         SliverToBoxAdapter(child: MyErrorWidget(error)),
-      AsyncValue(:final value) =>
-        _DataSliverList<T>(value.whereNonNull.toList())
+      _ => const DataListLoading(),
```

Comment:
What happened to `AsyncLoading`?

Reply:
That's the change that helped. This alone solves the problem. I got the idea from docs as well, from the pull to refresh guide. https://riverpod.dev/docs/case_studies/pull_to_refresh

What was the problem is the fact that after refresh the state changes to `AsyncLoading` and we showed the loading screen for a second.

It turns out through, even though the state is loading. State.value still holds previous value for us to use.

So I changed the pattern matching switch logic a bit.

Now if the async value have any value we show it. If it has error we show error and in all other cases (`_` means all unmatched previously things) we show loading screen. So the loading screen shows if we have async loading or async data with value=null